### DISCUSSION TITLE
Which font should we use for shields?

### DISCUSSION THREAD

**Discussion by claysmalley** [osm-americana/openstreetmap-americana#954] [General]
Seeking community feedback on shield fonts.

It's become clear from numerous discussions that we'll need to decide on a single font for shields to ensure consistency across platforms. Some prior pull requests have attempted to address this (#538, #544, #768), but questions remain unresolved about what style we want to emulate. Do we want to find a font visually similar to Highway Gothic? Or might another font better match the metrics we're looking for? I brainstormed a list of ideal characteristics for a shield font:

* condensed
* sans-serif
* numeral `1` has no bottom serif
* numeral `1` is at most ⅔ the width of other numerals (i.e. `111` fits in a two-digit shield)
* numeral `3` has two strokes, not three (no `>` shape on top)
* numeral `4` is closed
* numeral `7` has two strokes, not three (no serif on top)
* looks like it belongs on an American road atlas

I looked through [Google Fonts](https://fonts.google.com/) and found 8 fonts that meet most of these criteria. None of them are perfect, but I've made a summary of their pros and cons. Which ones are your favorites? Which ones should be removed from consideration? What stylistic direction should we aim for?

|Font|Links|Pros|Cons|
|-|-|-|-|
|[Antonio](https://fonts.google.com/specimen/Antonio)|[demo](https://preview.ourmap.us/pr/946/), [shield test](https://preview.ourmap.us/pr/946/shieldtest.html)||Missing Greek/Cyrillic|
|[Cabin Condensed](https://fonts.google.com/specimen/Cabin+Condensed)|[demo](https://preview.ourmap.us/pr/947/), [shield test](https://preview.ourmap.us/pr/947/shieldtest.html)|`0`, `1` and `8` are close matches to Highway Gothic; has National Geographic vibes|`2` and `4` are pointy; missing Greek/Cyrillic|
|[Encode Sans Condensed](https://fonts.google.com/specimen/Encode+Sans+Condensed)|[demo](https://preview.ourmap.us/pr/948/), [shield test](https://preview.ourmap.us/pr/948/shieldtest.html)||`4` is open; missing Greek/Cyrillic|
|[Fira Sans Extra Condensed](https://fonts.google.com/specimen/Fira+Sans+Extra+Condensed)|[demo](https://preview.ourmap.us/pr/949/), [shield test](https://preview.ourmap.us/pr/949/shieldtest.html)||`1` a little too wide, `4` is open|
|[Overpass](https://fonts.google.com/specimen/Overpass)|[demo](https://preview.ourmap.us/pr/950/), [shield test](https://preview.ourmap.us/pr/950/shieldtest.html)|Closest match to Highway Gothic|Not condensed; missing Greek|
|[Pathway Gothic One](https://fonts.google.com/specimen/Pathway+Gothic+One)|[demo](https://preview.ourmap.us/pr/951/), [shield test](https://preview.ourmap.us/pr/951/shieldtest.html)||Banner text is very narrow; missing Greek/Cyrillic|
|[Roboto Condensed](https://fonts.google.com/specimen/Roboto+Condensed)|[demo](https://preview.ourmap.us/pr/952/), [shield test](https://preview.ourmap.us/pr/952/shieldtest.html)|Close match to Noto for internationalization|`1` too wide (tabular numerals)|
|[Ubuntu Condensed](https://fonts.google.com/specimen/Ubuntu+Condensed)|[demo](https://preview.ourmap.us/pr/953/), [shield test](https://preview.ourmap.us/pr/953/shieldtest.html)||`1` too wide (tabular numerals)|
|[Oswald](https://fonts.google.com/specimen/Oswald) (suggested by @1ec5)|[demo](https://preview.ourmap.us/pr/955/), [shield test](https://preview.ourmap.us/pr/955/shieldtest.html)||Missing Greek|
|[Roboto Condensed](https://fonts.google.com/specimen/Roboto+Condensed) (flattened proportional nums)|[demo](https://preview.ourmap.us/pr/964/), [shield test](https://preview.ourmap.us/pr/964/shieldtest.html)|Close match to Noto for internationalization| |
|[Noto Sans Condensed](https://fonts.google.com/noto/specimen/Noto+Sans)  (flattened proportional nums)|[demo](https://preview.ourmap.us/pr/963/), [shield test](https://preview.ourmap.us/pr/963/shieldtest.html)|Matches rest of map style; seamless internationalization| |

**Comment by zekefarwell** [comment]
Wow!  Thank you for putting all of this together.  That is good list of ideal characteristics.  I would add one more:

- The tail of the 6 and 9 should point off at an angle rather than looping around too much

Like this: 
<img width="90" alt="Screenshot 2023-10-07 at 2 34 28 PM" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/cf3c478b-9944-4228-a808-6e80bb176613">

Not like this: 
<img width="76" alt="Screenshot 2023-10-07 at 2 34 10 PM" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/c3e514d8-b937-4f15-b912-7f1e4374aed5">

**Reply by ZeLonewolf** [reply]
Strong agree.  On that basis, I would strike "Antonio" and "Pathway Gothic" off the list of contenders.

**Reply by 1ec5** [reply]
I disagree that this needs to be a strict requirement – we aren’t designing signage for low-light conditions. But Pathway Gothic’s tails are rather extreme, so I would still shy away from it on that count.

**Reply by zekefarwell** [reply]
I also don't think it needs to be a strict requirement, just one of the ideal characteristics to aim for.  I don't expect we'll end up with a font that meets all these ideals, but I care a bit more about this detail than some of the others.  It's also not a binary choice.  For example Roboto Condensed has tails that curve more than Highway Gothic's (which are nearly straight):
<img width="48" alt="Screenshot 2023-10-07 at 5 31 04 PM" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/c224c16f-04a5-450b-9b65-69efbef52dfd">

But it's a closer match than Pathway Gothic's more pronounced scoop:

<img width="41" alt="Screenshot 2023-10-07 at 5 43 52 PM" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/3c8f712d-fd3f-41f3-ab7b-8410e25b8bc3">

**Reply by 1ec5** [reply]
I think the problem with the tails on Pathway Gothic is that they make the glyphs too dense at a small size. That font isn’t really designed for fine print. If a font manages to keep the density to a minimum even as it curves the tails, then I care a lot less about the tails.

**Reply by zekefarwell** [reply]
Yes overly dense is a good description of the curvier 6 and 9 tails.  I find it can easily make 6, 9, and 8 all looks too similar.  Oswald is an improvement over Pathway but I think less of a curve makes the numerals more distinct and easier to parse at a glance. 

<img width="24" alt="698-oswald" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/aa1cbb68-14f4-445a-9c03-52258a910ed8">

<img width="25" alt="698-roboto" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/34ff5ae5-ccfc-4f29-9ed3-217df98ced83">

**Reply by 1ec5** [reply]
I think I could get used to the 6s and 9s in either Roboto Condensed and Noto Sans Condensed. The apertures on Roboto (also on 2 and 5) are slightly more closed, which lends a more formal appearance. But it’s very much in the area of diminishing returns at this point.

**Comment by zekefarwell** [comment]
Overpass looks great but the argument for a condensed font is strong so I think this one should probably be eliminated (perhaps Red Hat will sponsor a condensed variant in the future). 

Of the other options Cabin Condensed, Ubuntu Condensed, and Roboto Condensed look the best to me (in order of preference).  They each have minor drawbacks, but I don't find them very noticeable at the font size we are using.

I find the drawbacks of Fira Sans Extra Condensed and Encode Sans Condensed a bit more noticeable, but these two seem like they would also work fine.

Pathway Gothic One and Antonio are the ones I like the least.  They both deviate from Highway Gothic numeral shapes much more noticeably.

**Reply by claysmalley** [reply]
> (perhaps Red Hat will sponsor a condensed variant in the future).

I can see their headquarters out my window. Should I go over and knock? 🤔

**Reply by zekefarwell** [reply]
[overpassfont.org](https://overpassfont.org/), so proud of its eight weights, true italics, and monospace!  Wouldn't you guys just love to add condensed to that list? 😉

**Reply by 1ec5** [reply]
Ironically, Highway Gothic comes in plenty of pitches, but Overpass is the only reproduction that has a half-decent character repertoire.

**Reply by 1ec5** [reply]
I think it’s fine if the font deviates from Highway Gothic numeral shapes to some extent – these are icons on a map, not signs. However, Antonio, Encode Sans, and Fira Sans are all too heavy. At these small sizes, we need a bold font weight, or _maybe_ a medium, but not a heavy or black font weight.

**Reply by claysmalley** [reply]
Font weight is now 500 for Antonio, Encode Sans and Fira Sans.

**Comment by ZeLonewolf** [comment]
We'll still need to deal with some residual pixel pushing, but I was interested to note some shield width issues we'll also have to resolve (example from #947):
![image](https://github.com/ZeLonewolf/openstreetmap-americana/assets/3254090/a709b3dd-d078-42e0-b04f-af09ddcd3c31)

**Reply by claysmalley** [reply]
The pixels we push depend on which font we choose, so I plan on adjusting all the text constraints in a future font-changing PR.

**Reply by 1ec5** [reply]
We already have extreme shield width problems with nonrectangular shields: #834. You just haven’t been noticing it as much with the less common fonts on your system. Regardless, we need to make sure that two-digit heater shields (Interstate, U.S., circle, California, etc.) have a roughly 1:1 aspect ratio. This has been the primary motivation for a condensed font.

**Reply by 1ec5** [reply]
Part of the problem is that the constraint code wants to scale the text to fit the shield, rather than the other way around. Automatically scaled text makes sense at the extremes, and for some shield designs perhaps, but there’s no good reason for the 1- and 2-digit shields to have differently sized text, or for two 2-digit shields to have differently sized text as in the screenshot above.

**Reply by zekefarwell** [reply]
I agree.  Sizing shield blanks to fit around 2, 3, and maybe 4 digits at a specified font and font size will give a more consistent looking result.  Shrinking the font size to fit when the ref gets longer than that makes sense though.

**Comment by 1ec5** [comment]
I assume this font will also apply to banner text, so we’ll need to consider how a variety of banners look with each option. In particular, four-letter banners still need to fit over two-digit shields.

**Reply by 1ec5** [reply]
Antonio, Cabin, Encode, and Pathway all lack Cyrillic and Greek glyphs. Although we can rely on automatic browser font fallbacks, this will result in some discrepancies in Eastern Europe and Central Asia where Latin letters coexist with Cyrillic and Greek letters on shields; all three scripts have similar enough letterforms that it’ll look somewhat jarring.

**Comment by 1ec5** [comment]
The 0 in Cabin Condensed is too wide. We want a variable-pitch font so that 1 can be a bit narrower, but I don’t think we want 0 to be wider than derivative glyphs like 8.

**Reply by claysmalley** [reply]
As pretty as Cabin Condensed looks, it's hard to work with when numeral widths are all over the place.
 
![image](https://github.com/ZeLonewolf/openstreetmap-americana/assets/1732117/353de3eb-f79f-4112-908e-cec63eb9a310)

**Reply by zekefarwell** [reply]
I hadn't noticed the wider zero.  That is a good point.

**Reply by Scott5114** [reply]
Cabin Condensed has an open license. If the only issue with it is the 0, it wouldn't be too hard to tweak it in FontForge to be the same width as the 8. Of course, that would mean Americana would have to host and maintain the modified font files, rather than simply pointing to Google's version.

**Reply by 1ec5** [reply]
I think it would be a great idea to host whichever Web font we ultimately decide on, since we’re already hosting the SDF font for map labels in [osm-americana/fontstack66](https://github.com/osm-americana/fontstack66/). There isn’t much performance benefit to using a common CDN for the fonts anyways. That said, I still think a grotesque sans-serif would fit the overall map’s look and feel and any fallback fonts much better than Cabin Condensed, attractive as it would be for certain labels.

**Reply by zekefarwell** [reply]
I agree.  Self hosting the chosen font is definitely the way to go these days.  Pulling directly from Google Fonts is great for these initial comparisons and testing though.

**Comment by 1ec5** [comment]
I’d like to add [Oswald](https://fonts.google.com/specimen/Oswald?preview.text=MA123456789-0EW%20ALT%20BUS%20TRCK&stroke=Sans+Serif) for consideration: an understated sans-serif, condensed, not too heavy, full complement of Latin/Cyrillic/Greek glyphs, narrow 1, standard-width 0. The 6 and 9 glyphs curve around, but the tails aren’t exaggerated. My only qualm is that the R has too high a bridge.

<img src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/0d410427-f855-46fe-a625-b19c9576170a" width="202" alt="Oswald">

**Reply by claysmalley** [reply]
Added!

**Reply by claysmalley** [reply]
Oswald on Google Fonts doesn't appear to have Greek. I guess they'll have to join the likes of Armenian and Georgian in jarring font changes.

**Reply by 1ec5** [reply]
Ah, I missed that. I think Greek would be nice and am surprised by its omission, but it’s a lower priority than Cyrillic. At least Greece is consistent about using Greek letters in route numbers within its borders, unlike some countries that use Cyrillic alphabets.

Oswald’s Cyrillic might not be a runaway success, judging from this critique: googlefonts/OswaldFont#21. Most of the shortcomings in that article would probably only become apparent in display typography, at larger sizes. The З/3 similarity would ordinarily be concerning, but hopefully no highway system is using З in conjunction with numerals, just as no jurisdiction would do that with vehicle registration plates.

Needless to say, I would advocate for using this font only for shields and not for other labeling needs.

**Reply by zekefarwell** [reply]
I'm liking Oswald but the 500 weight seems to be on the heavy side compared with the other fonts we're looking at.  I'm wondering if the 400 weight would be a better match to the shields.  It seems to already be slightly bold.

**Comment by pnorman** [comment]
I loaded them up in tabs and switched between them. I immediately removed Pathway Gothic One from consideration.

Encode Sans Condensed feels too light to match the look which Americana has been going with of replicating the shield outlines. You don't need to replicate the font - particularly since all the shields would have different fonts on the road - but the shields are designed to have a heavier font on them than this one. On some of the less pictorial ones, it looks okay.

Similarly, the proportions on Antonio seem wrong.

I consistently go with Roboto Condensed as my choice. Maybe this is influenced by seeing a lot of maps with Noto Sans. Being close to Noto Sans is both an advantage and disadvantage. I could see reasons to want a different font on shields than other text, but some shields are going to have to fall back to the wider coverage of Noto and you want those to look similar.

I also noted one of the US:AR shields is blue, while all the others are white.

**Reply by 1ec5** [reply]
> I also noted one of the US:AR shields is blue, while all the others are white.

Good eye! You probably spotted [Arkansas Highway 980](https://wiki.aaroads.com/wiki/Arkansas_Highway_980).

**Reply by 1ec5** [reply]
I also find Roboto Condensed to be the best fit for the style’s aesthetic. The only problem is the extremely generous tracking on the 1. It’s as if the font defaults to tabular numerals. Does it also include proportional numerals that we could enable by setting [`font-variant-numeric`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-numeric) to `proportional-nums`?

**Reply by zekefarwell** [reply]
I agree.  Roboto Condensed does seem to be quite a good fit.  The wide 1 does make 111 look a bit awkward, but this doesn't seem like a huge problem to me.  If it can be tightened up with a variant that would be an improvement though.

**Reply by pnorman** [reply]
I don't see the wide 1 as a problem. I would expect signs with the same number of digits to be the same size, which would naturally lead to wider letter spacing if it's all 1's

**Reply by claysmalley** [reply]
Roboto Condensed does have a proportional variant, but `font-variant-numeric` doesn't seem to be exposed in [CanvasRenderingContext2D](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font). Is there another way to do this in an HTML canvas?

**Reply by 1ec5** [reply]
> I don't see the wide 1 as a problem. I would expect signs with the same number of digits to be the same size, which would naturally lead to wider letter spacing if it's all 1's

It’s essentially a kerning issue; there’s a very apparent gap around the 1 that isn’t present among other numerals. In my opinion, it’s a more noticeable departure from signage than a curled 6 or 9 would be. Standards such as the MUTCD specifically call for 1 to get less horizontal space than any other numeral, and most print maps use proportional numerals as well. Tabular numerals are for aligning figures in a table column, but there’s less need for alignment among shields.

Monospaced numerals:

<img width="259" alt="Ảnh chụp Màn hình 2023-10-08 lúc 00 55 19" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/92317515-3a6e-4963-838f-7cdee151fdad">

Proportional numerals:

<img width="216" alt="Clark County Route 215" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/1d466bd4-2a32-4fce-adc9-444360e23485">
<img width="277" alt="U.S. Route 421" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/dba0627a-e1e1-4e78-a0e7-5d0f31418980">
<img width="223" alt="Highways 11, 12, and 21" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/27369fd4-b064-4e5b-ae1b-5f63478cb293">
<img width="292" alt="U.S. Route 231 and State Roads 68 and 161" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/4d6213bb-4fa1-41f2-848f-bb5f9f57f5e8">
<img width="276" alt="Kentucky Routes 212 and 1334" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/4ce2e31c-2ee4-460e-947b-3f3d329ca7d6">

**Reply by 1ec5** [reply]
> Roboto Condensed does have a proportional variant, but `font-variant-numeric` doesn't seem to be exposed in [CanvasRenderingContext2D](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font). Is there another way to do this in an HTML canvas?

The [`font`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font) property accepts CSS [`font`](https://developer.mozilla.org/en-US/docs/Web/CSS/font) syntax, which is shorthand for among other things [`font-variant`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant), which in turn is shorthand for [`font-variant-numeric`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-numeric). However, I haven’t tried it out myself.

**Reply by 1ec5** [reply]
Roboto Condensed’s full-width 1 is not nearly as jarring if we lighten the font:

400 | 700
----|----
![Shield test at 400 weight](https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/93ff64f2-152f-4d1b-be23-4fc6a7de7c0b) | ![Shield test at 700 weight](https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/5a4a62ed-8a96-4ddf-87cc-255e46d5be3a)
[<img src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/5cbd4f9d-dead-4756-9b30-c129359860a6" width="400" alt="Louisiana Highways 13, 3116, and 3123 at 400 weight">](https://preview.ourmap.us/pr/952/#map=14/30.46125/-92.41407) | [<img src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/383c0e2c-bc75-42a5-b4d1-daf01c808b96" width="400" alt="Louisiana Highways 13, 3116, and 3123 at 700 weight">](https://preview.ourmap.us/pr/952/#map=14/30.46125/-92.41407)
[<img src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/d8fb295c-91d2-4181-9dd5-f3c171ba10f1" width="400" alt="Interstate 10 and U.S. Routes 165 and 171 at 400 weight">](https://preview.ourmap.us/pr/952/#map=9/30.4008/-93.081) | [<img src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/8a9ec168-17c6-4aeb-aa54-5a7b40404a3b" width="400" alt="Interstate 10 and U.S. Routes 165 and 171 at 700 weight">](https://preview.ourmap.us/pr/952/#map=9/30.4008/-93.081)

A normal font weight differs from the signs, but it really evokes the majority of print maps, particularly older ones. (For some reason, I tend to see boldfaced route shields more commonly in maps of Kentucky than any other state.)

**Reply by 1ec5** [reply]
>> Roboto Condensed does have a proportional variant, but `font-variant-numeric` doesn't seem to be exposed in [CanvasRenderingContext2D](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font). Is there another way to do this in an HTML canvas?
> 
> The [`font`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font) property accepts CSS [`font`](https://developer.mozilla.org/en-US/docs/Web/CSS/font) syntax, which is shorthand for among other things [`font-variant`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant), which in turn is shorthand for [`font-variant-numeric`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-numeric). However, I haven’t tried it out myself.

I guess this isn’t supported. 😅

<img src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/32e72f63-9033-4bf1-a61e-0c8133502399" width="400" alt="Fine print">

I missed the caveat that [`font`](https://developer.mozilla.org/en-US/docs/Web/CSS/font) only allows the `font-variant` keywords from CSS 2.1 (`normal` and `small-caps`).

**Reply by zekefarwell** [reply]
> Roboto Condensed’s full-width 1 is not nearly as jarring if we lighten the font

Hmm... I think we're looking at two different fonts here?
<img width="144" alt="Screenshot 2023-10-08 at 7 24 12 PM" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/81a9f0ff-9e79-4757-aba1-ef907480bcfb">
<img width="140" alt="Screenshot 2023-10-08 at 7 23 58 PM" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/b7f54e21-b64e-4c48-a17a-2720828ff3e6">
<img width="151" alt="Screenshot 2023-10-08 at 7 23 39 PM" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/40d98281-0dfb-4dcf-b27a-047b38fb79c4">

**Reply by 1ec5** [reply]
I had to edit my post after the fact when realizing I was on the wrong branch. Sorry about that!

**Reply by 1ec5** [reply]
Oh, wait, that’s the same font loading issue as https://github.com/osm-americana/openstreetmap-americana/pull/955#issuecomment-1751888674. 🤦

**Reply by zekefarwell** [reply]
Yeah or similar behavior anyway.  I've had a heck of a time being sure I'm really seeing the intended font when testing.

**Comment by zekefarwell** [comment]
> Do we want to find a font visually similar to Highway Gothic? Or might another font better match the metrics we're looking for?

I had gotten the impression that there was quite a bit of enthusiasm for something visually similar to Highway Gothic so that's what I've been using as a benchmark.  This style faithfully adapts real world shield designs in great detail so choosing a font that is also faithful (or at least similar) to the real world signs makes sense to me (American signs anyway, but hey the style is named Americana after all).  If I've misread this enthusiasm, then I also feel fine choosing a font that simply works well for this use case.

**Reply by 1ec5** [reply]
I think we toe the line of literalism without crossing into it. We do go the extra mile in identifying edge cases, such as the blue Arkansas Highway 980 shield, and we’ve gone the easy route by aligning the color palette to the MUTCD. However, I think users will forgive us for straying from Highway Gothic while remaining within the norms of map publishing. There are a lot of free fonts that fit the bill, just not as many on Google Fonts.

I’ve yet to encounter a print map that uses Highway Gothic on its shields, other than in an inset explaining the Interstate system with a large illustration of a shield. For ordinary markers on the map, I’d say that the norm is a medium-to-bold, often condensed grotesque or neo-grotesque typeface. I’ve seen some examples of geometric typefaces in transportation maps, particularly older ones, but humanist sans-serifs or slab serifs would be confined to tourist recreational maps.

**Reply by zekefarwell** [reply]
Just to be clear, Highway Gothic itself has never been under consideration.  Overpass is inspired by Highway Gothic but designed for small text size and screen display so would work quite well on a map if it met our requirements by having a condensed variant.  We're clearly going to use a different font than this.  The question is simply if we're aiming for a font with somewhat similar shapes to Highway Gothic or not.

**Reply by claysmalley** [reply]
While it would be faithful to the real-world designs to adopt a Highway Gothic lookalike, we end up straying significantly from real-world shields anyway in our simplification process. Plus, we're going for an American road atlas aesthetic. Some atlases use a Highway Gothic lookalike, some don't.

**Comment by zekefarwell** [comment]
I assume we haven't been considering Noto Sans (except as an internationalization fallback) for lack of a condensed variant.  However [Noto Sans Display](https://fonts.google.com/noto/specimen/Noto+Sans+Display/tester) is a variable width font with a range from extra condensed to regular.  Perhaps this could work for us?
 
<img width="572" alt="Screenshot 2023-10-08 at 10 19 15 AM" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/0348f84d-f06f-4b71-a16a-e29b2a3404bd">

**Reply by zekefarwell** [reply]
Unfortunately it appears that variable features of variable fonts are still not usable in canvas although a discussion about it was started five years ago:
https://github.com/whatwg/html/issues/3571

It looks like a very complex workaround is possible but that doesn't seem like a good path to take:
https://stackoverflow.com/questions/65670959/variable-fonts-in-html5-canvas

**Reply by 1ec5** [reply]
Noto Sans Display is designed for larger text like headlines. Noto Sans would be sufficient at these small sizes. The variable font stuff is for developer convenience only; a specific Noto Sans Condensed font is available for download from the [Noto Dashboard](https://notofonts.github.io/#latin-greek-cyrillic). Per https://github.com/ZeLonewolf/openstreetmap-americana/discussions/954#discussioncomment-7222509, we should host this font on GitHub Pages as part of either this repository or fontstack66. The fontstack66 repository already pulls in google-font-installer as a dependency to ensure we’re getting all the latest bug fixes and updates to the Noto fonts.

**Reply by zekefarwell** [reply]
Oh good there is a condensed variant.  Not sure why I was having a hard time finding it.  This one seems worth considering as well then.

**Reply by 1ec5** [reply]
Noto Sans has proportional numerals but also defaults to monospaced numerals. Here’s Noto SemiCondensed SemiBold:

<img width="102" alt="Noto SemiCondensed SemiBold" src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/1231218/1a0146ca-8984-47a4-abd0-b4b7bc8147c6">

**Comment by ZeLonewolf** [comment]
On thing that's missing is a look at some paper maps.  Below are 1958 and 2011 AAA paper map screen shots, respectively. I was surprised to note that the fonts do not replicate physical shields but rather are differently chosen for map display.  In particular, the Highway Gothic features - the clipped 6 and 9, the no-bottom-serif 1, and the snowman 8 - are not present on the paper map. On that basis, it seems fine to use a font that doesn't clip the tail of the 6 and 9.  We might also consider whether per-network font weights are something we want to try to do.  Note how much bolder the Interstate shield font is compared to the other routes around it.

![2023-10-08 20 46 48](https://github.com/ZeLonewolf/openstreetmap-americana/assets/3254090/fb74ea31-2fbc-4371-85ea-f04ca1811cd9)
![2023-10-08 20 47 27](https://github.com/ZeLonewolf/openstreetmap-americana/assets/3254090/e9684c89-466b-4828-be0d-e9ca59583bc7)

**Reply by zekefarwell** [reply]
> On thing that's missing is a look at some paper maps

In fairness, @1ec5 did post a number of paper map samples in https://github.com/ZeLonewolf/openstreetmap-americana/discussions/954#discussioncomment-7224139, but this branching conversation is hard to keep track of 😅.  Rand McNally maps have short tails on the 6 and 9, but the 1 is also narrow.  I see why you all consider that important.  It does make the best use of space.

<img src="https://github.com/ZeLonewolf/openstreetmap-americana/assets/281482/d7532a32-71a3-4cfe-8d68-73427ea964c5" width="700px"/>

**Reply by 1ec5** [reply]
> We might also consider whether per-network font weights are something we want to try to do. Note how much bolder the Interstate shield font is compared to the other routes around it.

Some print maps make light text on a dark background bolder than dark text on a light background. I suspect this is just to keep the numerals from becoming illegible if the ink separates or smudges.

**Comment by zekefarwell** [comment]
This is unrelated to font family choice, but I find I'm enjoying lighter weight text on the banners.  Feels more readable for such small text and helps the shield stand out more.

| Encode 500 - lighter | Fira 500 - heavier |
|--------|--------|
|  <img width="349" alt="encode-lighter" src="https://github.com/osm-americana/openstreetmap-americana/assets/281482/59703839-2757-4c21-b441-6cd3f733bb5a"> |  <img width="349" alt="fira-heavier" src="https://github.com/osm-americana/openstreetmap-americana/assets/281482/4bbab78c-a09e-4d09-9f79-08158ba08762"> |

**Comment by claysmalley** [comment]
I'm thoroughly enjoying Noto Sans. It matches the rest of the style, and it makes internationalization easy. This one's a winner to me.

Next steps, if we want to go forward with Noto Sans:
- [ ] Create repository for `.woff` files, including the following fonts:
   - Noto Sans Condensed Medium
   - Noto Sans Armenian Condensed Medium
- [ ] Pull request to change fonts and adjust metrics

**Comment by ZeLonewolf** [comment]
This is now resolved, thanks everyone for the hard work!
